### PR TITLE
Update docs to use ngx-markdown

### PR DESCRIPTION
**Pull request by shanmukhateja**

### CODE REVIEW

File: demo/angular.json [hunk 1]
```diff
@@ -31,7 +31,9 @@
               "node_modules/datatables.net-select-dt/css/select.dataTables.css",
               "node_modules/materialize-css/dist/css/materialize.css",
               "node_modules/highlight.js/styles/github.css",
-              "src/styles.css"
+              "node_modules/prism-themes/themes/prism-material-dark.css",
```
Comment (l-lin) [inline demo/angular.json:34]: __suggestion__: Is PrismJS a syntax highlighting library? What's the difference with highlight.js? If it's better, we can remove the latter :)

File: demo/src/app/base-demo/base-demo.component.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<div class="section banner">
+  <div class="container">
+    <div class="row">
+      <div class="col s12 m9">
+        <h1 class="header center-on-small-only">Zero configuration</h1>
```
Comment (l-lin) [inline demo/src/app/base-demo/base-demo.component.html:5]: __thoughts__: Reminder to have this removed or configurable.

File: demo/angular.json (hunk 1 above)
Comment (shanmukhateja) [reply demo/angular.json:34]: Yes, PrismJS is a syntax highlighting library. ngx-markdown uses PrismJS for highlighting blocks of code we use. It also provides the "Copy to Clipboard" functionality when download from npm.

Yes, I do plan on removing highlight.js and its angular wrapper. I'll need to delete CJS warning suppression section in `angular.json`.

File: demo/src/app/base-demo/base-demo.component.html (hunk 2 above)
Comment (shanmukhateja) [reply demo/src/app/base-demo/base-demo.component.html:5]: Yes, this block will be configurable. I seemed to have forgotten about this.

File: demo/src/assets/docs/basic/server-side-angular-way/source-html.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+```typescript
```
Comment (l-lin) [inline demo/src/assets/docs/basic/server-side-angular-way/source-html.md:1]:
```suggestion
```html
```

File: demo/src/app/app.component.html [hunk 4]
```diff
@@ -126,11 +130,16 @@ <h3>
 <footer class="page-footer">
     <div class="footer-copyright">
         <div class="container">
-            Designed and built with &lt;3 by
-            <a class="grey-text text-lighten-4" href="https://github.com/l-lin">Louis LIN</a>
-            <p class="right">
-                <iframe class="right github-stars" src="https://ghbtns.com/github-btn.html?user=l-lin&amp;repo=angular-datatables&amp;type=watch&amp;count=true&amp;size=medium"
-                    allowtransparency="true" frameborder="0" scrolling="0" width="84" height="20"></iframe>
+            <p>
+                <span class="red-text text-darken-2"> Designed and built with &lt;3 by
```
Comment (l-lin) [inline demo/src/app/app.component.html:134]: __question__: I'm not sure why you opt to use red here as I found having white text on blue background more readable. :confused:

File: demo/src/app/app.component.html [hunk 5]
```diff
@@ -126,11 +130,16 @@ <h3>
...
+                    <a class="grey-text text-darken-4" href="https://github.com/l-lin"><u>Louis LIN</u></a> and
+                    contributors.
+                </span>
+                <span class="right">
+                    <iframe class="right github-stars"
```
Comment (l-lin) [inline demo/src/app/app.component.html:139]: __suggestion__: I think we can remove this iframe as the link to the github project is already present in the sidebar.

**Review by l-lin — CHANGES_REQUESTED** [review]
It's way better than better! Really nice work! 

I like the home page and I noticed you updated the FAQ, nice :+1:

File: demo/src/assets/docs/basic/server-side-angular-way/source-html.md (hunk 3 above)
Comment (shanmukhateja) [reply demo/src/assets/docs/basic/server-side-angular-way/source-html.md]: Fixed.

File: demo/src/app/app.component.html (hunk 4 above)
Comment (shanmukhateja) [reply demo/src/app/app.component.html]: The red text was a placeholder for me like a TODO. I reverted it and slightly increased font-weight to improve text visibility in larger screens.

File: demo/src/app/app.component.html (hunk 5 above)
Comment (shanmukhateja) [reply demo/src/app/app.component.html]: Fixed.

**Review by l-lin — APPROVED** [review]

**Pull request merged**